src/crate/client/sqlalchemy/dialect.py

```diff
@@ -190,6 +198,13 @@ def do_rollback(self, connection):
         pass
 
     def connect(self, host=None, port=None, *args, **kwargs):
+
+        # Optionally set server version through SQLAlchemy connection string
+        # query parameter explicitly.
+        if 'cratedb-version' in kwargs:
+            forced_version = LooseVersion(kwargs.pop('cratedb-version'))
+            self.server_version_info = tuple(forced_version.version)
```

Comment:
Why do we need to set the cratedb version via connection string? I don't think this is something we should expose for users.

If this is required for testing, wouldn't it be possible to mock the cursor accordingly?

Reply:
You are right, its purpose was only to cover some test cases. Exposing it for users indeed is a bad idea. I've submitted d91d088c7972 to handle that differently. I hope you like it better.

Also, when removing support for CrateDB < 1.0.1, we can get rid of this adjustment completely.

Reply:
The adjustment became altogether obsolete with #424.